src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.spec.ts

```diff
@@ -1,37 +1,109 @@
 import { ToastrModule } from 'ngx-toastr';
 import { SharedModule } from '~/app/shared/shared.module';
-
-import { RgwMultisiteZoneFormComponent } from './rgw-multisite-zone-form.component';
-import { configureTestBed } from '~/testing/unit-test-helper';
+import { RgwZone } from '../models/rgw-multisite';
 
 describe('RgwMultisiteZoneFormComponent', () => {
   let component: RgwMultisiteZoneFormComponent;
   let fixture: ComponentFixture<RgwMultisiteZoneFormComponent>;
-
-  configureTestBed({
-    imports: [
-      SharedModule,
-      ReactiveFormsModule,
-      RouterTestingModule,
-      HttpClientTestingModule,
-      ToastrModule.forRoot()
-    ],
-    providers: [NgbActiveModal],
-    declarations: [RgwMultisiteZoneFormComponent]
-  });
+  let rgwZoneService: RgwZoneService;
+  let rgwZoneServicePoolsSpy: jasmine.Spy;
+  let rgwZoneServiceSpy: jasmine.Spy;
 
   beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        SharedModule,
+        ReactiveFormsModule,
+        RouterTestingModule,
+        HttpClientTestingModule,
+        ToastrModule.forRoot()
+      ],
+      providers: [NgbActiveModal],
+      declarations: [RgwMultisiteZoneFormComponent]
+    }).compileComponents();
+
     fixture = TestBed.createComponent(RgwMultisiteZoneFormComponent);
     component = fixture.componentInstance;
+    rgwZoneService = TestBed.inject(RgwZoneService);
+
+    rgwZoneServicePoolsSpy = spyOn(rgwZoneService, 'getPoolNames').and.returnValue(of([]));
+    rgwZoneServiceSpy = spyOn(rgwZoneService, 'get');
+
+    rgwZoneServiceSpy.and.returnValue(
+      of({
+        placement_pools: [
+          {
+            key: 'default-placement',
+            val: {
+              storage_classes: {
+                STANDARD: {
+                  data_pool: 'standard-data-pool',
+                  compression_type: 'gzip'
+                }
+              },
+              index_pool: 'index-pool',
+              data_extra_pool: 'extra-data-pool'
+            }
+          }
+        ]
+      })
+    );
+
+    component.info = {
+      parent: {
+        data: {
+          name: 'zonegroup2',
+          placement_targets: [
+            { name: 'default-placement', tags: [], storage_classes: ['STANDARD'] }
+          ],
+          default_placement: 'default-placement'
+        }
+      },
+      data: {
+        name: 'zone2',
+        parent: 'zonegroup2',
+        is_default: true,
+        is_master: true,
+        endpoints: ['http://192.168.100.100:80'],
+        access_key: 'zxcftyuuhgg',
+        secret_key: 'Qwsdcfgghuiioklpoozsd'
+      }
+    };
+
+    component.zone = new RgwZone();
+    component.zone.name = component.info.data.name;
+    component.action = 'edit';
+
     fixture.detectChanges();
+
+    component.getZonePlacementData('default-placement');
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should set correct values in the form on edit', () => {
```

Comment:
why dont we add one for create as well :) ?
Meanwhile I can test this too

Reply:
Added for create as well